inference.py

```diff
@@ -31,6 +31,9 @@ def run_inference(args):
     # load model
     if "llama" in args.model_name_or_path.lower(): # special case for Facebook's LLaMA model
         llm = LLAMA(args)
+    # FIXME: Exception for encoder-decoder models is generic at the moment.
+    elif args.is_encoder_decoder:
+        llm = LLM(args, is_encoder_decoder=True)
```

Comment:
Perhaps we should simply handle `is_encoder_decoder` in the `LLM.__init__` method. Then a `llm = LLM(args)` would suffice.